cmake/FindTreesitter.cmake

```diff
@@ -8,22 +8,5 @@ add_library(treesitter INTERFACE)
 target_include_directories(treesitter SYSTEM BEFORE INTERFACE ${TREESITTER_INCLUDE_DIR})
 target_link_libraries(treesitter INTERFACE ${TREESITTER_LIBRARY})
 
-# TODO(lewis6991): remove when min TS version is 0.20.9
-list(APPEND CMAKE_REQUIRED_INCLUDES "${TREESITTER_INCLUDE_DIR}")
-list(APPEND CMAKE_REQUIRED_LIBRARIES "${TREESITTER_LIBRARY}")
-check_c_source_compiles("
-#include <tree_sitter/api.h>
-int
-main(void)
-{
-  TSQueryCursor *cursor = ts_query_cursor_new();
-  ts_query_cursor_set_max_start_depth(cursor, 32);
-  return 0;
-}
-" TS_HAS_SET_MAX_START_DEPTH)
 list(REMOVE_ITEM CMAKE_REQUIRED_INCLUDES "${TREESITTER_INCLUDE_DIR}")
 list(REMOVE_ITEM CMAKE_REQUIRED_LIBRARIES "${TREESITTER_LIBRARY}")
```

Comment:
Remove, this is only needed since we previously added `TREESITTER_INCLUDE_DIR` to `CMAKE_REQUIRED_INCLUDES`. Same with `TREESITTER_LIBRARY`.